im.srain.Srain.json

```diff
@@ -0,0 +1,146 @@
+{
+  "app-id": "im.srain.Srain",
+  "runtime": "org.gnome.Platform",
+  "runtime-version": "3.26",
+  "sdk": "org.gnome.Sdk",
+  "command": "srain",
+  "rename-desktop-file": "Srain.desktop",
+  "rename-appdata-file": "Srain.appdata.xml",
+  "rename-icon": "srain",
+  "copy-icon": true,
+  "finish-args": [
+    "--share=ipc",
+    "--device=all",
+    "--filesystem=home:ro",
```

Comment:
What's this for? As a GTK app it can easily use the portal and not need extra permissions

Reply:
It's needed so that users can select images and other files to upload. I'm not very sure on what the *portal* is, would it require modifying the application?

Reply:
For a GTK app it requires that the app uses [GTKFileChooserNative](https://developer.gnome.org/gtk3/stable/gtk3-GtkFileChooserNative.html) rather than GTKFileChooser.

Portals are ways for apps to poke holes in the sandbox under user control e.g. the app only gets access to the specific file the user wants to upload rather than any file in the home directory (which could include ssh keys etc.) Feel free to have more of a read: http://docs.flatpak.org/en/latest/introduction.html#how-it-works